.github/workflows/coverage.yml

```diff
@@ -0,0 +1,149 @@
+name: coverage
+
+on:
+  schedule:
+    - cron: '0 1 * * *'
+
+jobs:
+  gcc13_coverage_build:
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/4c-multiphysics/4c-dependencies-ubuntu24.04:28cfb9d0
+      options: --user root --env OMPI_ALLOW_RUN_AS_ROOT=1 --env OMPI_ALLOW_RUN_AS_ROOT_CONFIRM=1
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v4
+      - name: Check docker hash
+        uses: ./.github/actions/compute-and-check-dependencies-hash
+      - uses: ./.github/actions/build_4C
+        with:
+          cmake-preset: docker_coverage
+          build-targets: full
+          build-directory: ${{ github.workspace }}/build
+          use-ccache: "false"
+      - uses: ./.github/actions/upload_4C_build
+        with:
+          build-directory: ${{ github.workspace }}/build
+          retention-days: 1
+
+  gcc13_coverage_test:
+    needs: gcc13_coverage_build
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/4c-multiphysics/4c-dependencies-ubuntu24.04:28cfb9d0
+      options: --user root --env OMPI_ALLOW_RUN_AS_ROOT=1 --env OMPI_ALLOW_RUN_AS_ROOT_CONFIRM=1
+    strategy:
+      fail-fast: true
+      matrix:
+        # we need to make very fine grained chunks to avoid running out of storage
+        test-chunk: ['1,20', '21,40', '41,60', '61,80', '81,100', '101,120', '121,140', '141,160', '161,180',
```

Comment:
I guess this is very brittle due to the small chunk sizes? I am fine with merging but we might need to generate these chunks dynamically.

Reply:
Do you intend to use the automatic chunking from your other PR here?

Reply:
Yes, once it delivers reliable results after #132 and #129 are merged.